### PR TITLE
10

### PR DESCRIPTION
**Pull request by karlrivkpmg**

### CODE REVIEW

**Comment by karlrivkpmg** [comment]
Accidentally did some of the hosting prep in this branch will fix conflicts when possible

File: server/__tests__/app.test.js [hunk 1]
```diff
@@ -417,15 +417,122 @@ describe('8. GET /api/users', () =>{
...
+        })
+    })
+
+    test("status:400, given a bad category query", ()=>{
+        return request(app)
+        .get('/api/reviews?category=asdf')
+        .expect(400)
+        .then((response)=>{
+            expect(response.body.msg).toBe("Category does not exist");
+        })
```
Comment (Dutton1990) [inline server/__tests__/app.test.js:461]: This should be a 404 as new categories could be added and the data type is correct. No need to do a 400 for this one though for an incorrect data type

File: server/__tests__/app.test.js [hunk 2]
```diff
@@ -417,15 +417,122 @@ describe('8. GET /api/users', () =>{
...
+    })
+
+    test("status:400, given a bad sort_by query", ()=>{
+        return request(app)
+        .get('/api/reviews?sort_by=asdf')
+        .expect(400)
+        .then((response)=>{
+            expect(response.body.msg).toBe("Column does not exist");
+        })
```
Comment (Dutton1990) [inline server/__tests__/app.test.js:496]: 400 for this one is correct as we have full control of what columns we can sort by

File: server/models/model.boardgames.js [hunk 3]
```diff
@@ -12,15 +12,54 @@ exports.selectCategories = () =>{
...
+
+    if(category !== undefined){
+        if(allowedCategories.includes(category)){
+            queryValues.push(category);
+            sql += `WHERE reviews.category = $1 `;
+        }
+        else{
+            return Promise.reject({msg: "Category does not exist", status:400});
+        }
+
```
Comment (Dutton1990) [inline server/models/model.boardgames.js:55]: Really nice to check that the category exists instead of hard coding the values!

**Review by Dutton1990 — CHANGES_REQUESTED** [review]
Awesome, really nice testing suite for this. A suggested change above on one of them. 

There's another test you can add, there should be a category that exists but has no reviews attached to it. This should return 200 with an empty array. 

If you could make those changes then re-send the PR that would be great, cheers Karl! Great work overall

**Review by lewispricey — APPROVED** [review]
Great work Karl, thanks for adding in that extra test!

Good to merge 🚀

**Pull request merged**